src/pages/user/User.page.tsx

```diff
@@ -0,0 +1,22 @@
+import { Button } from "@components";
+import { useNavigate } from "react-router-dom";
+import { useAuth } from "@providers";
+
+const UserPage = () => {
+    const authProvider = useAuth();
+    const navigate = useNavigate();
+
+    const handleLogout = () => {
+        authProvider.logout();
+        navigate("/user/login", { replace: true });
```

Comment:
I think using navigate after a logout will result in conflicts with redirects that happens within the protected route group. Given that you should only be able to sign out in a protected group where you need authentication. The `ProtectedRoutes` component will redirect automatically once the user log outs.

I think the better pattern here is to pass a redirect string argument to `login()` and `logout()` methods provided by the `AuthProvider` then the `AuthProvider` can handle redirects.

Reply:
Either way, `ProtectedRoutes` should have already covered the redirecting once the user successfully logs out.